### PR TITLE
[no gbp] Golem mineral functionality tweaks/fixes

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

Fixes a couple of things people have pointed out about golem transformations.

- Diamond golems now stop being invisible when they attack or throw something, they also turn invisible a bit slower.
- Using a bluespace knot takes 2 seconds instead of 3 seconds.
- Bananium Golems only slip you if they are lying down.

In order to achieve that last one I refactored the slipperiness component to take an optional extra callback, and then killed a subtype of it which could be replaced with passing in a callback. I tested it and it seems to work the same as it used to.

These are largely how things were supposed to work and I just overlooked them.
I am sure this won't be the last PR of a similar vein while people try these out, provided that I actually hear anything they are saying about it.

## Why It's Good For The Game

Diamond golems shouldn't be able to attack you while remaining invisible and untargetable even if it is funny.
Clown golems aren't supposed to be able to slip you by swapping places with you even if it is funny.
The bluespace hand was basically just worse than using the crystal and not eating it, maybe still needs another buff after this one.

## Changelog

:cl:
fix: Diamond Golems can no longer attack or throw things and remain invisible.
fix: Bananium Golems are only slippery if you actually tread on them (aka: while they are resting).
balance: Golem bluespace teleportation is slightly quicker.
/:cl:

### CODE REVIEW

File: code/datums/components/slippery.dm [hunk 1]
```diff
@@ -28,12 +30,21 @@
...
-/datum/component/slippery/Initialize(knockdown, lube_flags = NONE, datum/callback/callback, paralyze, force_drop = FALSE, slot_whitelist)
+/datum/component/slippery/Initialize(
+	knockdown,
+	lube_flags = NONE,
+	datum/callback/on_slip_callback,
+	paralyze,
+	force_drop = FALSE,
+	slot_whitelist,
+	datum/callback/can_slip_callback,
+	)
```
Comment (san7890) [inline code/datums/components/slippery.dm:41]:
```suggestion
)
```

it's ugly but we should do it the right way

File: code/datums/components/slippery.dm [hunk 2]
```diff
@@ -49,11 +60,12 @@
 	if(ismovable(parent))
 		AddComponent(/datum/component/connect_loc_behalf, parent, default_connections)
 
-/datum/component/slippery/InheritComponent(datum/component/slippery/component, i_am_original, knockdown, lube_flags = NONE, datum/callback/callback, paralyze, force_drop = FALSE, slot_whitelist)
+/datum/component/slippery/InheritComponent(datum/component/slippery/component, i_am_original, knockdown, lube_flags = NONE, datum/callback/on_slip_callback, paralyze, force_drop = FALSE, slot_whitelist)
```
Comment (san7890) [inline code/datums/components/slippery.dm:63]: since initialize is multilined, should this be as well?

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/datums/components/slippery.dm [hunk 3]
```diff
@@ -49,11 +60,21 @@
 	if(ismovable(parent))
 		AddComponent(/datum/component/connect_loc_behalf, parent, default_connections)
 
-/datum/component/slippery/InheritComponent(datum/component/slippery/component, i_am_original, knockdown, lube_flags = NONE, datum/callback/callback, paralyze, force_drop = FALSE, slot_whitelist)
+/datum/component/slippery/InheritComponent(
+	datum/component/slippery/component,
+	i_am_original, knockdown,
+	lube_flags = NONE,
```
Comment (san7890) [inline code/datums/components/slippery.dm:66]:
```suggestion
	datum/component/slippery/component,
	i_am_original,
	knockdown,
	lube_flags = NONE,
```

**Comment by san7890** [comment]
c6486f43f5882c31912326c0f39fb40d107c3cad

it wasn't missing you just didn't multiline it right

**Comment by Jacquerel** [comment]
> [c6486f4](https://github.com/tgstation/tgstation/commit/c6486f43f5882c31912326c0f39fb40d107c3cad)
> 
> it wasn't missing you just didn't multiline it right

and we can see why multilining it right is important because as a result of doing it wrong i then thought it wasnt there 🤔

**Pull request merged**